docs/guides/network-profiler.md

```diff
@@ -0,0 +1,170 @@
+
+Following these steps will let you see how numerous rigidbody instances lead to substantial network traffic, helping address this in your own projects.
+
+## Setting up the profiler
+
+To diagnose and optimize bandwidth in your project, set up Normcore's profiler module in Unity's Profiler window. This will visualize network traffic data, making bandwidth issues easier to identify and address. Follow these steps:
+
+- Open Unity's Profiler window `Window > Analysis > Profiler`.
+- In the Profiler window, click on the `Profiler Modules` dropdown.
+- Select `Normcore` from the available profiler modules.
+
+<video width="100%" controls><source src={profilerSetUp} /></video> 
+
+### Profiler data
+
+With the Normcore profiler active, you'll see two key metrics in the Profiler window:
+- **Sent**: Data being sent from the client over time.
+- **Received**: Data being received by the client over time.
+
+## Profiling
+
+Pay attention to the graphs for sent and received data. Look for spikes and trends correlating with specific events or actions. For instance, if you see a jump in data when new rigidbody instances are created, those instances are likely contributing heavily to traffic.
+
+This video shows how sent data increases over time as new instances are created. The profiler reveals a pattern where each instance adds to the cumulative sent data, causing spikes.
+
+<video width="100%" controls><source src={profileUnlimitedInstances} /></video> 
+
+## Solution
+
+To optimize bandwidth, it's crucial to manage rigidbody instances effectively. Two common strategies are reusing instances and destroying those that are too far from the player or no longer contributing to the simulation. Implementing these strategies can significantly reduce transmitted data.
+
+In our sample, we have implemented a pooling mechanism that will create instances up to a certain limit and reuse the oldest instance when the instance limit is reached.
+
+```csharp
+using System.Collections.Generic;
+using Normal.Realtime;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class ShooterOptimized : MonoBehaviour {
+    // A queue to store basketballs
+    private Queue<GameObject> _basketballPool = new Queue<GameObject>();
+
+    // Called once per frame
+    private void Update() {
+        // Check if the left mouse button was pressed this frame
+        if (Mouse.current.leftButton.wasPressedThisFrame) ShootTenBasketballs();
+    }
+
+    // Shoot 10 basketballs
+    private void ShootTenBasketballs() {
+        for (var i = 0; i < 10; i++) Shoot();
+    }
+
+    // Shoot method to get an instance from the pool
+    private void Shoot() {
+        // Spawn a basketball in the same location as the shooter component
+        var basketball = InstantiateBasketballFromPool();
+        
+        // Set the velocity to a random forward/up velocity
+        basketball.GetComponent<Rigidbody>().velocity = basketball.transform.forward * 5f;
```

Comment:
basketball.transform.forward is always Vector3.forward right? Would it make sense to adjust the basketball rotation with a random range so they fire at slightly different angles?

Reply:
basketball.transform.forward at this point is transform.forward. I've added randomness on the position so they don't instantiate at the same exact place and "explode". We can add direction randomness, but I don't think it will add value to the sample.